### ISSUE TITLE
knife cookbook upload -o` does not parse absolute Windows paths as expected

### ISSUE THREAD

**Issue by dheerajd-msys** [chef/chef#10082]
## Description

`knife cookbook upload foo.bar -o C:\dir1\cookbooks` (with a variety of delimiters) parses the colon as a delimiter instead of as part of an absolute path on Windows. Through trial and error, the user was able to determine that a double colon could be used, or they could use a relative path after replacing forward slashes with backslashes. The user needs to be able to use an Azure DevOps variable that returns an absolute path, preferably without modifying the variable value much. It's possible there's a best practice, but it doesn't appear to be documented so this may be a documentation "bug".
## Related issue
https://github.com/chef/customer-bugs/issues/203
## Replication Case

Run `knife cookbook upload foo.bar -o C:\somepath\cookbooks` (or some variant using documented string delimiters)
## Client Output

```

```
## Expected Behavior
Should upload the designated cookbook using an absolute path on a Windows box

## Actual Behavior
":" is parsed as a path delimiter so the command fails when it tries to find two paths instead of one absolute path.

**Comment by antima-gupta** [comment]
@tas50 @lamont-granquist Currently for uploading multiple cookbooks from multiple paths, we use `:` to separate paths where cookbooks in.Code is here.It is failing on Windows system as in windows absolute path would be like `Drive Name:\directory`.

To fix this issue we can use `,` to separate the paths Or can use both `,` and `:`,  so that user can provide anyone option for split path and mention in the doc only for windows use , .

Please let me know your suggestions on this.

**Comment by phiggins** [comment]
A good way to fix this while maintaining backwards compatibility would be to use `File::PATH_SEPARATOR`.

On my macbook it is `:`:
```
irb(main):003:0> File::PATH_SEPARATOR
=> ":"
```

On a windows VM, it is `;`:
```
irb(main):001:0> File::PATH_SEPARATOR
=> ";"
```

**Comment by dheerajd-msys** [comment]
This issue is fixed via #10146 . 
Closing it now. Thanks @antima-gupta for working on it.
